src/v/cluster/partition_balancer_planner.cc

```diff
@@ -1598,16 +1590,12 @@ partition_balancer_planner::plan_actions(
       ctx, ctx.decommissioning_nodes, change_reason::node_decommissioning);
 
     if (ctx.config().mode == model::partition_autobalancing_mode::continuous) {
-        if (ctx.num_nodes_in_maintenance == 0) {
```

Comment:
there is an expectation that the balancer will not move partitions to the nodes in maintenance mode expressed in https://github.com/redpanda-data/redpanda/issues/12069. Let's make sure we are aligned on that we are currently going to move some partitions to nodes in maintenance mode

Reply:
Synced with @mattschumpert and decided to keep it out of scope; partition balancer already handles offline nodes and from its perspective there is no difference between a node being offline and a node being offline and in maintenance mode. On the other hand if a node is maintenance mode then it doesn't necessary mean it's offline.

Reply:
Hmm, but a node going in maintenance mode is often (always?) a precursor to the node going offline soon.